frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -1606,16 +1606,45 @@ class SegmentsView extends React.Component<Props, State> {
     if (expandedGroups) {
       this.setExpandedGroupsFromSet(expandedGroups);
     }
+  };
+
+  handleSearchSelect = (selectedElement: SegmentHierarchyNode) => {
+    this.maybeExpandParentGroup(selectedElement);
     // As parent groups might still need to expand, we need to wait for this to finish.
     setTimeout(() => {
       if (this.tree.current) this.tree.current.scrollTo({ key: selectedElement.key });
     }, 50);
     const isASegment = "color" in selectedElement;
     if (isASegment) {
       this.onSelectSegment(selectedElement);
+    } else {
+      if (this.props.visibleSegmentationLayer == null) return;
+      Store.dispatch(
+        setSelectedSegmentsOrGroupAction(
+          [],
+          selectedElement.id,
+          this.props.visibleSegmentationLayer?.name,
+        ),
+      );
     }
   };
 
+  handleSelectAllMatchingSegments = (allMatches: SegmentHierarchyNode[]) => {
+    if (this.props.visibleSegmentationLayer == null) return;
+    const allMatchingSegmentIds = allMatches.map((match) => {
+      this.maybeExpandParentGroup(match);
+      return match.id;
+    });
+    Store.dispatch(
+      setSelectedSegmentsOrGroupAction(
+        allMatchingSegmentIds,
+        null,
+        this.props.visibleSegmentationLayer.name,
+      ),
+    );
+    this.tree.current?.scrollTo({ key: allMatches[0].key });
```

Comment:
Could you please give this a small timeout like in lines 1614-1616? Sometimes the browser is kinda slow due to some workload resulting in the segment groups not being all expanded and thus `allMatches[0].key` might still not be shown in the list. In such a case the antd lib ignored the scrollTo call.
```suggestion
 setTimeout(() => {
    this.tree.current?.scrollTo({ key: allMatches[0].key });
     }, 50);
```